pkg/motor/resource.go

```diff
@@ -0,0 +1,111 @@
+package motor
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	st "github.com/sonr-io/sonr/x/schema/types"
+)
+
+// SchemaCallback is a callback used when a schema CID is fetched
+type SchemaCallback func(*st.SchemaDefinition, error)
+type motorResources struct {
+	mu sync.Mutex
+
+	config            *client.Client
+	schemaQueryClient st.QueryClient
+
+	whatIsStore map[string]*st.WhatIs
+	schemaStore map[string]*st.SchemaDefinition
+}
+
+func newMotorResources(config *client.Client, schemaQueryClient st.QueryClient) *motorResources {
+	return &motorResources{
+		config:            config,
+		schemaQueryClient: schemaQueryClient,
+
+		whatIsStore: make(map[string]*st.WhatIs),
+		schemaStore: make(map[string]*st.SchemaDefinition),
+	}
+}
+
+/// StoreWhatIs fetches
+func (r *motorResources) StoreWhatIs(whatIs *st.WhatIs, cb SchemaCallback) {
+	callback := func(s *st.SchemaDefinition, e error) {
+		r.mu.Unlock()
+		cb(s, e)
+	}
+
+	go func() {
+		r.mu.Lock()
+
+		r.whatIsStore[whatIs.Did] = whatIs
+
+		if whatIs.Schema == nil {
+			callback(nil, fmt.Errorf("WhatIs '%s' has no schema", whatIs.Did))
```

Comment:
nit: errors should be constants in the package

Reply:
Yes I agree. One thing with constants is that we can't easily include details like the DID. I created #505 to deal with this.

Reply:
sounds good to me and i agree. resolving